lib/netext/grpcext/conn.go

```diff
@@ -0,0 +1,305 @@
+	state := h.vu.State()
+	tags := getTags(ctx)
+	switch s := stat.(type) {
+	case *grpcstats.OutHeader:
+		if state.Options.SystemTags.Has(metrics.TagIP) && s.RemoteAddr != nil {
+			if ip, _, err := net.SplitHostPort(s.RemoteAddr.String()); err == nil {
+				tags["ip"] = ip
+			}
+		}
+	case *grpcstats.End:
+		if state.Options.SystemTags.Has(metrics.TagStatus) {
+			tags["status"] = strconv.Itoa(int(status.Code(s.Error)))
+		}
+
+		mTags := map[string]string(tags)
+		sampleTags := metrics.IntoSampleTags(&mTags)
+		metrics.PushIfNotDone(ctx, state.Samples, metrics.ConnectedSamples{
+			Samples: []metrics.Sample{
+				{
+					Metric: state.BuiltinMetrics.GRPCReqDuration,
+					Tags:   sampleTags,
+					Value:  metrics.D(s.EndTime.Sub(s.BeginTime)),
+					Time:   s.EndTime,
+				},
+			},
+		})
+	}
+
+	// (rogchap) Re-using --http-debug flag as gRPC is technically still HTTP
+	if state.Options.HTTPDebug.String != "" {
+		logger := state.Logger.WithField("source", "http-debug")
+		httpDebugOption := state.Options.HTTPDebug.String
+		DebugStat(logger, stat, httpDebugOption)
+	}
+}
+
+// DebugStat prints debugging information based on RPCStats.
+func DebugStat(logger logrus.FieldLogger, stat grpcstats.RPCStats, httpDebugOption string) {
+	switch s := stat.(type) {
+	case *grpcstats.OutHeader:
+		logger.Infof("Out Header:\nFull Method: %s\nRemote Address: %s\n%s\n",
+			s.FullMethod, s.RemoteAddr, formatMetadata(s.Header))
+	case *grpcstats.OutTrailer:
+		if len(s.Trailer) > 0 {
+			logger.Infof("Out Trailer:\n%s\n", formatMetadata(s.Trailer))
+		}
+	case *grpcstats.OutPayload:
+		if httpDebugOption == "full" {
+			logger.Infof("Out Payload:\nWire Length: %d\nSent Time: %s\n%s\n\n",
+				s.WireLength, s.SentTime, formatPayload(s.Payload))
+		}
+	case *grpcstats.InHeader:
+		if len(s.Header) > 0 {
+			logger.Infof("In Header:\nWire Length: %d\n%s\n", s.WireLength, formatMetadata(s.Header))
+		}
+	case *grpcstats.InTrailer:
+		if len(s.Trailer) > 0 {
+			logger.Infof("In Trailer:\nWire Length: %d\n%s\n", s.WireLength, formatMetadata(s.Trailer))
+		}
+	case *grpcstats.InPayload:
+		if httpDebugOption == "full" {
+			logger.Infof("In Payload:\nWire Length: %d\nReceived Time: %s\n%s\n\n",
+				s.WireLength, s.RecvTime, formatPayload(s.Payload))
+		}
+	}
+}
+
+func formatMetadata(md metadata.MD) string {
+	var sb strings.Builder
+	for k, v := range md {
+		sb.WriteString(k)
+		sb.WriteString(": ")
+		sb.WriteString(strings.Join(v, ", "))
+		sb.WriteRune('\n')
+	}
+
+	return sb.String()
+}
+
+func formatPayload(payload interface{}) string {
+	msg, ok := payload.(proto.Message)
+	if !ok {
+		// check to see if we are dealing with a APIv1 message
+		msgV1, ok := payload.(protov1.Message)
+		if !ok {
+			return ""
+		}
+		msg = protov1.MessageV2(msgV1)
+	}
+
+	marshaler := prototext.MarshalOptions{
+		Multiline: true,
+		Indent:    "  ",
+	}
+	b, err := marshaler.Marshal(msg)
+	if err != nil {
+		return ""
+	}
+	return string(b)
+}
+
+type ctxKeyTags struct{}
```

Comment:
you haven't deleted these from the js package, it seems: https://github.com/grafana/k6/blob/grpcext/js/modules/k6/grpc/tags.go